src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -511,6 +519,31 @@ public UUID getFirstSourceDatasetIdFromSnapshotId(
         .orElseThrow(() -> new DatasetNotFoundException("Source dataset for snapshot not found"));
   }
 
+  /**
+   * @param snapshotId snapshot UUID
+   * @param userReq authenticated user
+   * @return SAM-derived snapshot policies attributed to the user, including workspace resolution
+   *     where applicable.
+   */
+  public PolicyResponse retrieveSnapshotPolicies(
```

Comment:
thanks for moving this into the service.  Makes more sense here than the controller